.github/workflows/update-dependencies.yaml

```
@@ -0,0 +1,115 @@
+name: update-dependencies
+on:
+  workflow_call:
+    inputs:
+      component:
+        type: string
+        required: true
+      write-lock:
+        type: string
+        required: true
+      label:
+        type: string
+        required: true
+
+jobs:
+  init:
+    runs-on: [ "self-hosted", "builder" ]
+    outputs:
+      metadata: ${{ steps.init.outputs.metadata }}
+      parameters: ${{ steps.init.outputs.parameters }}
+    steps:
+      - id: init
+        name: Init Workflow
+        uses: hawk-ai-aml/github-actions/workflow-init@master
+        with:
+          component: ${{ inputs.component }}
+          profile: hawk
+          update-kustomize: false
+          
+  lock-dependencies:
+    needs: [ init ]
+    runs-on: [ "self-hosted", "builder" ]
+    outputs:
+      result: ${{ steps.commit-and-push.outputs.result }}
+    steps:
+      - id: init
+        name: Init Job
+        uses: hawk-ai-aml/github-actions/job-init@master
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ fromJson(needs.init.outputs.parameters).java-version }}
+          distribution: temurin
+
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.REPO_ACCESS_PAT }}
+          ref: update-dependencies
+
+      - name: Fetch and reset update-dependencies to master
+        run: |
+          git fetch origin master
+          git reset --hard origin/master
+
+      - name: Perform dependency resolution and write new lockfiles
+        run: |
+          ${{ inputs.write-lock }}
+          git status
+      - name: Commit, push, create pull request
+        id: commit-and-push
+        run: |
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "github-actions@hawk.ai"
+          git add .
+          
+          if ! git diff --quiet HEAD; then
+            git commit -m "build(gradle): Update gradle.lockfile"
+            git push --force origin update-dependencies
+            echo "result=success" >> $GITHUB_OUTPUT
+          else 
+            echo "result=skipped" >> $GITHUB_OUTPUT
+            echo no dynamic dependencies updates
```

Echo needs to be put into "".
```suggestion
            echo "no dynamic dependencies updates"
```